doc/dev/rbd-diff.rst

```diff
@@ -67,3 +67,80 @@ End
 ---
 
 - u8: 'e'
+
+
+Header
+~~~~~~
+
+"rbd diff v2\\n"
+
+Metadata records
+~~~~~~~~~~~~~~~~
+
+Every record has a one byte "tag" that identifies the record type,
+followed by length of data, and then some other data.
+
+Metadata records come in the first part of the image.  Order is not
+important, as long as all the metadata records come before the data
+records.
+
+In v2, we have the following metadata in each section:
+(1 Bytes) tag.
+(8 Bytes) length.
```

Comment:
I think it's safe to use a 32bit length -- the only thing that requires 64bit is the write/zero tag, but since the maximum object size is 32M, they technically only require 32bit lengths as well (and should be updated to reflect that).